### PR TITLE
Project: Do not use a sorted set for `scopeDependencies` / `scopeNames`

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

Part of #6235.

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/GoDep.kt [hunk 1]
```diff
@@ -185,7 +185,7 @@ class GoDep(
                     vcs = VcsInfo.EMPTY,
                     vcsProcessed = projectVcs,
                     homepageUrl = "",
-                    scopeDependencies = sortedSetOf(scope)
+                    scopeDependencies = mutableSetOf(scope)
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/GoDep.kt:188]: Commit message:
- Probably put `scopeDependencies` into backticks (as you do so for `scopeNames` in the next commit)
- Typo in "unk*n*own"

File: model/src/main/kotlin/utils/DependencyGraphBuilder.kt [hunk 2]
```diff
@@ -219,10 +218,11 @@ class DependencyGraphBuilder<D>(
      * Return a set of all the scope names known to this builder that start with the given [prefix]. If [unqualify] is
      * *true*, remove this prefix from the returned scope names.
      */
-    fun scopesFor(prefix: String, unqualify: Boolean = true): SortedSet<String> {
-        val qualifiedScopes = scopeMapping.keys.filter { it.startsWith(prefix) }
-        val scopes = qualifiedScopes.takeUnless { unqualify } ?: qualifiedScopes.map { it.substring(prefix.length) }
-        return scopes.toSortedSet()
+    fun scopesFor(prefix: String, unqualify: Boolean = true): Set<String> {
```
Comment (sschuberth) [inline model/src/main/kotlin/utils/DependencyGraphBuilder.kt:221]: Commit message: Typo in "prece~e~ding" (the stroked through "e" does not really render well)

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: analyzer/src/main/kotlin/managers/GoDep.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (fviernau) [reply analyzer/src/main/kotlin/managers/GoDep.kt:188]: The backticks don't fit into the 72 chars.

[the remark this answers is not in the document]
Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/GoDep.kt]: > The backticks don't fit into the 72 chars.

IIRC we've configured commitlint to allow 75 chars in the title like in the body (even if GitHub itself suggests 72 chars).

**Review by sschuberth — APPROVED** [review]

**Pull request merged**